aws/resource_aws_elasticsearch_domain.go

```diff
@@ -634,6 +683,12 @@ func resourceAwsElasticSearchDomainUpdate(d *schema.ResourceData, meta interface
 		input.VPCOptions = expandESVPCOptions(s)
 	}
 
+	if d.HasChange("cognito_options") {
+		options := d.Get("cognito_options").([]interface{})
+		s := options[0].(map[string]interface{})
```

Comment:
This line looks like it can cause a panic if `cognito_options` is removed from the configuration since `options` will have 0 elements. We should perform a length check and set the parameter to nil/empty/disabled as required, e.g.

```go
if d.HasChange("cognito_options") {
  // default to disabling (change as necessary! this could also be handled in expandESCognitoOptions)
  input.CognitoOptions = &elasticsearch.CognitoOptions{
    Enabled: aws.Bool(false),
  }
  // only enable if provided
  if v, ok := d.GetOk("cognito_options"); ok && len(v.([]interface{})) > 0 {
    m := v.([]interface{})[0].(map[string]interface{})
    input.CognitoOptions = expandESCognitoOptions(m)
  }
}
```

I'd recommend adding a second TestStep to the acceptance test that covers trying to remove the `cognito_options` to ensure we're okay here. 👍